core/lib/multivm/src/tracers/validator/vm_latest/mod.rs

```diff
@@ -81,6 +83,50 @@ impl<H: HistoryMode> ValidationTracer<H> {
                             called_address,
                         ));
                     }
+                    // If this is a call to the timestamp asserter, extract the function arguments and store them in ValidationTraces.
+                    // These arguments are used by the mempool for transaction filtering. The call data length should be 68 bytes:
+                    // a 4-byte function selector followed by two U256 values.
+                    let address = self.timestamp_asserter_params.as_ref().map(|x| x.address);
+                    if Some(called_address) == address
+                        && far_call_abi.memory_quasi_fat_pointer.length == 68
+                    {
+                        let calldata_page = get_calldata_page_via_abi(
+                            &far_call_abi,
+                            state.vm_local_state.callstack.current.base_memory_page,
+                        );
+                        let calldata = memory.read_unaligned_bytes(
+                            calldata_page as usize,
+                            far_call_abi.memory_quasi_fat_pointer.start as usize,
+                            68,
+                        );
+
+                        if calldata[..4] == TIMESTAMP_ASSERTER_FUNCTION_SELECTOR {
+                            let start = U256::from_big_endian(
+                                &calldata[calldata.len() - 64..calldata.len() - 32],
+                            );
+                            let end = U256::from_big_endian(&calldata[calldata.len() - 32..]);
+
+                            let params = self.timestamp_asserter_params.as_ref().unwrap();
+
+                            if end.as_u32() - start.as_u32() < params.min_range_sec {
```

Comment:
* `end` is guaranteed to be greater than `start`, this is enforced in the contract, i.e.
```
        require(start < block.timestamp && end > block.timestamp, "Timestamp is out of range");
```
* added an extra bit of logic to avoid overflows - values are capped at `u64::MAX` / `u32::MAX`